src/Uno.Toolkit.RuntimeTests/Tests/ItemsRepeaterExtensionTests.ISupportIncrementalLoading.cs

```diff
@@ -0,0 +1,248 @@
+
+		return sv;
+	}
+
+	[TestMethod]
+	[RequiresFullWindow]
+	[RunsOnUIThread]
+	[DataRow(Orientation.Vertical)]
+	[DataRow(Orientation.Horizontal)]
+	public async Task When_Incremental_Load_ShouldStop(Orientation orientation)
+	{
+		const int BatchSize = 25;
+		var source = new InfiniteSource<int>(async start =>
+		{
+			await Task.Delay(25);
+			return Enumerable.Range(start, BatchSize).ToArray();
+		});
+		var SUT = new Grid() { Height = 210, Width = 210, VerticalAlignment = VerticalAlignment.Bottom };
+		var ir = SetupIncrementalLoadingItemsRepeater(source, orientation);
+		var sv = SetupScrollViewer(ir);
+		SUT.Children.Add(sv);
+
+		await UnitTestUIContentHelperEx.SetContentAndWait(SUT);
+
+		await Task.Delay(1000);
+		var initial = GetCurrenState();
+
+		(double? hOffset, double? vOffset) = orientation switch
+		{
+			Orientation.Vertical => ((double?)null, 10_000),
+			Orientation.Horizontal => (10_000, default),
+			_ => throw new NotSupportedException()
+		};
+
+		// scroll to bottom
+		sv.ChangeView(hOffset, vOffset, null, disableAnimation: true);
+		await Task.Delay(500);
+		await UnitTestsUIContentHelper.WaitForIdle();
+		var firstScroll = GetCurrenState();
+
+		// Has'No'MoreItems
+		source.HasMoreItems = false;
+
+		// scroll to bottom
+		sv.ChangeView(hOffset, vOffset, null, disableAnimation: true);
+		await Task.Delay(500);
+		await UnitTestsUIContentHelper.WaitForIdle();
+		var secondScroll = GetCurrenState();
+
+		Assert.AreEqual(BatchSize * 1, initial.LastLoaded, "Should start with first batch loaded.");
+		Assert.AreEqual(BatchSize * 2, firstScroll.LastLoaded, "Should have 2 batches loaded after first scroll.");
+		Assert.IsTrue(initial.LastMaterialized < firstScroll.LastMaterialized, "No extra item materialized after first scroll.");
+		Assert.AreEqual(BatchSize * 2, secondScroll.LastLoaded, "Should still have 2 batches loaded after first scroll since HasMoreItems was false.");
+		Assert.AreEqual(BatchSize * 2 - 1, secondScroll.LastMaterialized, "Last materialized item should be the last from 2nd batch (50th/index=49).");
+
+		(int LastLoaded, int LastMaterialized) GetCurrenState() =>
+		(
+			source.LastIndex,
+			Enumerable.Range(0, source.LastIndex).Reverse().FirstOrDefault(x => ir.TryGetElement(x) != null)
+		);
+	}
+
+	[TestMethod]
+	[RequiresFullWindow]
+	[RunsOnUIThread]
+	[DataRow(Orientation.Vertical)]
+	[DataRow(Orientation.Horizontal)]
+	public async Task When_Incremental_Load_IsLoading(Orientation orientation)
+	{
+		const int BatchSize = 25;
+		var source = new InfiniteSource<int>(async start =>
+		{
+			await Task.Delay(1000);
+			return Enumerable.Range(start, BatchSize).ToArray();
+		});
+		var SUT = new Grid() { Height = 210, Width = 210, VerticalAlignment = VerticalAlignment.Bottom };
+		var ir = SetupIncrementalLoadingItemsRepeater(source, orientation);
+		var sv = SetupScrollViewer(ir);
+		SUT.Children.Add(sv);
+
+		await UnitTestUIContentHelperEx.SetContentAndWait(SUT);
+		await Task.Delay(2000);
```

Comment:
you are to start using arbitrary wait time here
maybe you could use a `const int LoadingTime = 1000` with `Task.Delay(LoadingTime * 2)` to make it more clear